scene/gui/file_dialog.cpp

```diff
@@ -1007,13 +1033,213 @@ void FileDialog::_update_drives(bool p_select) {
 
 bool FileDialog::default_show_hidden_files = false;
 
+Array FileDialog::_get_options() const {
+	Array out;
+	for (const FileDialog::Option &opt : options) {
+		Dictionary dict;
+		dict["name"] = opt.name;
+		dict["values"] = opt.values;
+		dict["default"] = selected_options.has(opt.name) ? (int)selected_options[opt.name] : opt.default_idx;
+		out.push_back(dict);
+	}
+	return out;
+}
+
+void FileDialog::_opt_changed_chk(bool p_pressed, const String &p_name) {
+	if (selected_options.has(p_name)) {
+		selected_options[p_name] = p_pressed;
+	}
+}
+
+void FileDialog::_opt_changed_itm(int p_idx, const String &p_name) {
+	if (selected_options.has(p_name)) {
+		selected_options[p_name] = p_idx;
+	}
+}
+
+void FileDialog::_update_option_controls() {
+	if (!options_dirty) {
+		return;
+	}
+	options_dirty = false;
+
+	while (grid_options->get_child_count(false) > 0) {
+		Node *child = grid_options->get_child(0);
+		grid_options->remove_child(child);
+		child->queue_free();
```

Comment:
Pretty sure you can just do `memfree()`.

Reply:
Every similar instance seems to be using `queue_free()`.